src/main/webapp/app/exercises/programming/assess/programming-assessment-dashboard/programming-assessment-dashboard.component.html

```diff
@@ -0,0 +1,180 @@
+                            {{ submission.participation.submissions.length }}
+                        </ng-container>
+                    </td>
+                    <td>
+                        <span *ngIf="submission.latestResult && submission.latestResult!.score != null"> {{ submission.latestResult!.score }} % </span>
+                    </td>
+                    <td>
+                        {{ submission.durationInMinutes }}
+                        {{ 'artemisApp.assessment.dashboard.minutes' | translate }}
+                    </td>
+                    <td>{{ assessmentTypeTranslationKey(submission.latestResult) | translate }}</td>
+                    <td>
+                        <span *ngIf="submission.latestResult && submission.latestResult!.assessor">{{ submission.latestResult!.assessor!.name }}</span>
+                    </td>
+                    <td *ngFor="let item of [].constructor(numberOfCorrectionrounds); let correctionRound = index">
+                        <span *jhiHasAnyAuthority="['ROLE_ADMIN', 'ROLE_INSTRUCTOR', 'ROLE_TA']">
+                            <!-- exercise without exam-->
+                            <span *ngIf="exercise.type === ExerciseType.PROGRAMMING && exercise.course && submission">
+                                <a *ngIf="submission.results.length == 0" [class.disabled]="busy" [routerLink]="getAssessmentLink(submission)" class="btn btn-success btn-sm mb-1">
+                                    <fa-icon [fixedWidth]="true" [icon]="'folder-open'"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.assess' | translate }}
+                                </a>
+                                <a
+                                    *ngIf="submission.results.length == 1 && submission.latestResult.completionDate"
+                                    [class.disabled]="busy"
+                                    [routerLink]="getAssessmentLink(submission)"
+                                    class="btn btn-primary btn-sm mb-1"
+                                >
+                                    <fa-icon [fixedWidth]="true" [icon]="'folder-open'"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.open' | translate }}
+                                </a>
+                                <a
+                                    *ngIf="submission.results.length == 1 && !submission.latestResult.completionDate"
+                                    [class.disabled]="busy"
+                                    [routerLink]="getAssessmentLink(submission)"
+                                    class="btn btn-warning btn-sm mb-1"
+                                >
+                                    <fa-icon [fixedWidth]="true" [icon]="'folder-open'"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.continue' | translate }}
+                                </a>
+                                <a
+                                    (click)="cancelAssessment(submission)"
```

Comment:
This doesn't use a router link but fires an `click` event, so it should remain a button.

Reply:
ah yea you are right :) I changed all buttons and did indeed overlook this case. I'll change it back.